src/mongocrypt-marking.c

```diff
@@ -457,13 +457,31 @@ _mongocrypt_fle2_placeholder_to_insert_update_ciphertext (
 
    _mongocrypt_buffer_from_iter (&value, &placeholder->v_iter);
 
-   if (!_mongocrypt_fle2_placeholder_common (kb,
-                                             &common,
-                                             &placeholder->index_key_id,
-                                             &value,
-                                             true, /* derive tokens using counter */
-                                             placeholder->maxContentionCounter,
-                                             status)) {
+   int64_t contentionFactor = 0;
+   if (placeholder->maxContentionCounter < 0) {
+      CLIENT_ERR ("maxContentionCounter must be non-negative, got: %" PRId64,
+                  placeholder->maxContentionCounter);
+      goto fail;
+   }
+   if (placeholder->maxContentionCounter > 0) {
+      /* Choose a random contentionFactor in the inclusive range [0,
+       * placeholder->maxContentionCounter] */
+      if (!_mongocrypt_random_int64 (crypto,
+                                     placeholder->maxContentionCounter + 1,
```

Comment:
If `maxContentionCounter` is `INT64_MAX`, that'll overflow this parameter, which will be UB, and then later a divide-by-zero with the `%` operator.

Reply:
Scratch the divide-by-zero, since that's checked, but still might overflow.